pyproject.toml

```diff
@@ -19,7 +19,7 @@ maintainers = [
   { name = "Samuel Larkin", email = "Samuel.Larkin@nrc-cnrc.gc.ca" },
 ]
 readme = "README.md"
-requires-python = ">=3.10, <3.12"
+requires-python = ">=3.10, <3.13"
```

Comment:
This is a major change, it should not be hidden in an unrelated PR like this. It's fantastic news if we're now compatible with 3.12, but that should be in its own PR flagging it as a new feature, which fixes #329, and adding 3.12 to the classifiers in pyproject.toml.

Reply:
And now that we're compatible with three different Python versions, we should add matrix testing to keep them all exercised, which is another good reason to splice it out of this PR.